pkg/controller/plan/kubevirt.go

```diff
@@ -1519,6 +1519,10 @@ func (r *KubeVirt) secret(vmRef ref.Ref) (object *core.Secret, err error) {
 				"-") + "-",
 		},
 	}
+	if populator {
+		object.Data = r.Source.Secret.Data
+		return
+	}
```

Comment:
how did CDI use the secret without having the data set?

Reply:
CDI has it with different key names (few lines below it calls the secret builder):
```
object.StringData = map[string]string{
		"accessKeyId": string(in.Data["user"]),
		"secretKey":   string(in.Data["password"]),
}
```
they also have configmap that holds the CA.

Reply:
ok, so let's be more explicit here - it's not the populator being used or not but whether we expect the secret to be a clone of the provider's secret or to have the information spread across the secret and the config map?

Reply:
> another thing is that we always provided the provider secret name instead the new one we create.

yes, @yaacov saw that when trying to migrate to a different namespace
we need to change the CI to exercise this flow

Reply:
#158 without the changes in the namespaces (using `default`) checks it out. this is how evgeny saw it and reached out.